src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -431,16 +445,55 @@ internal void ParseSolution()
 
             ParseFileHeader();
 
+            HashSet<string> slnFilter = null;
+            if (_solutionFilterFile != null)
+            {
+                slnFilter = new HashSet<string>();
+                string filterFile = File.ReadAllText(_solutionFilterFile);
+                int projectsInd = filterFile.IndexOf('[', filterFile.IndexOf("\"projects\":"));
```

Comment:
Use `System.Text.Json` to parse the file. VS uses a JSON library so it'll be tolerant of various quirks and we should be similarly tolerant.

Reply:
What do you have to do to include System.Text.Json? `using System.Text.Json` didn't work.